provider/pkg/provider/provider.go

```diff
@@ -1368,6 +1369,21 @@ func (k *kubeProvider) Diff(
 		}
 	}
 
+	if metadata.ReplaceUnready(newInputs) {
+		switch newInputs.GetKind() {
+		case "Job":
+			jobChecker := states.NewJobChecker()
+			job, err := clients.FromUnstructured(newInputs)
+			if err == nil {
+				jobChecker.Update(job)
+				if !jobChecker.Ready() {
+					hasChanges = pulumirpc.DiffResponse_DIFF_SOME
+					replaces = append(replaces, `.metadata.annotations["pulumi.com/replaceUnready"]`)
+				}
+			}
+		}
```

Comment:
Should we add a default case to warn that it is currently only supported for jobs?

Reply:
Sure, that's a good idea.